### PR TITLE
coqchk: add marshal checking for `Module with` and VM data

### PR DESCRIPTION
**Pull request by SkySkimmer**
With this PR the only remaining Any
should be for ignored values (eg libobject stuff), except for the
Const_val case of structured_constant as it's an arbitrary
value.

### CODE REVIEW

File: checker/check.ml [hunk 1]
```diff
@@ -354,7 +354,7 @@ let intern_from_file ~intern_mode (dir, f) =
       let opaque_csts = marshal_in_segment ~validate ~value:Values.v_univopaques ~segment:seg_univs f ch in
       let tasks = marshal_in_segment ~validate ~value:Values.(Opt Any) ~segment:seg_tasks f ch in
       let table = marshal_in_segment ~validate ~value:Values.v_opaquetable ~segment:seg_opaque f ch in
-      let vmlib = marshal_in_segment ~validate ~value:Any ~segment:seg_vmlib f ch in
+      let vmlib = marshal_in_segment ~validate ~value:Values.v_vmlib ~segment:seg_vmlib f ch in
```
Comment (ppedrot) [inline checker/check.ml:357]: This should be conditional on the vm flag being set.

Comment (SkySkimmer) [reply checker/check.ml:357]: Why?

Comment (SkySkimmer) [reply checker/check.ml:357]: Do you mean the whole marshal or just using v_vmlib vs Any?

Comment (ppedrot) [reply checker/check.ml:357]: Both, which is implied by the first one.

File: checker/check.ml [hunk 2]
```diff
@@ -353,7 +353,10 @@ let intern_from_file ~intern_mode (dir, f) =
       let opaque_csts = marshal_in_segment ~validate ~value:Values.v_univopaques ~segment:seg_univs f ch in
       let tasks = marshal_in_segment ~validate ~value:Values.(Opt Any) ~segment:seg_tasks f ch in
       let table = marshal_in_segment ~validate ~value:Values.v_opaquetable ~segment:seg_opaque f ch in
-      let vmlib = marshal_in_segment ~validate ~value:Values.v_vmlib ~segment:seg_vmlib f ch in
+      let vmlib = if enable_VM
+        then marshal_in_segment ~validate ~value:Values.v_vmlib ~segment:seg_vmlib f ch
+        else Vmlibrary.(export (set_path dir empty))
```
Comment (ppedrot) [inline checker/check.ml:358]: I think that rather than a dummy value, we should change the library_vm field to an option.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot run full ci

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**